packages/expo-server/src/vendor/environment/common.ts

```diff
@@ -33,10 +33,30 @@ interface EnvironmentInput {
 }
 
 export function createEnvironment(input: EnvironmentInput) {
+  let manifest: Manifest | null = null;
+
   return {
     async getRoutesManifest(): Promise<Manifest> {
       const json = await input.readJson('_expo/routes.json');
-      return initManifestRegExp(json as RawManifest);
+      manifest = initManifestRegExp(json as RawManifest);
+      return manifest;
+    },
+
+    beforeResponse(responseInit: any, route: any) {
```

Comment:
Do we have any reservations to moving this to `src/vendor/abstract.ts`? I don't really see this being different in any case, so moving it would be nice. My main concern here is that we'd like this to be consistent and the types of `responsInit` and `route` are consistent, but if we move it here, we're opening it up to changes across environments.

The main intention of the `before*` hooks was basically to allow the CLI to do specific things and to allow EAS Hosting to step in (since it needs additional response serialisation), so not sure we need to make this customisable and feed this back.

Reply:
No reservations as such, it just seemed like a convenient place to hook into the response lifecycle. Will address in a followup commit.